csrc/type.cpp

```diff
@@ -32,6 +32,31 @@ KernelIndexMode indexTypeToMode(DataType index_type) {
       : KernelIndexMode::INT64;
 }
 
+bool isWiderType(const DataType& ref, const DataType& type) {
```

Comment:
I think you missed the double->complex double and float->complex float cast?

Reply:
Good catch, resolved.